langchain/src/retrievers/self_query/utils.ts

```diff
@@ -16,3 +21,184 @@ export function isFilterEmpty(
   }
   return isObject(filter) && Object.keys(filter).length === 0;
 }
+
+type SupabaseFilterProps = keyof SupabaseFilter;
+
+export class ProxyParamsDuplicator {
+  duplicationAllowedOps: string[] = [
+    "eq",
+    "neq",
+    "lt",
+    "lte",
+    "gt",
+    "gte",
+    "like",
+    "ilike",
+    "or",
+    "in",
+    "contains",
+    "match",
+    "not",
+    "textSearch",
+    "filter",
+  ];
+
+  values: [string, string][] = [];
+
+  buildProxyHandler() {
+    const proxyHandler: ProxyHandler<SupabaseFilter> = {
+      get: (target, prop, receiver) => {
+        if (typeof target[prop as SupabaseFilterProps] === "function") {
+          return (...args: any[]) => {
```

Comment:
Just to cut down on the number of casts - can we type this as a `[string, string | unknown]` tuple or something similar

Reply:
Sure!